torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2364,6 +2363,38 @@ std::tuple<Tensor, Tensor> atan2_backward(const Tensor& grad, const Tensor& self
             output_mask[0] ? grad * other * recip : Tensor(),
             output_mask[1] ? grad * -self * recip : Tensor() };
 }
+std::tuple<Tensor, Tensor, Tensor> attn_backward(
+    const Tensor& grad_o,
+    const Tensor& grad_a,
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v,
+    const Tensor& a) {
+  auto grad_q = Tensor();
+  auto grad_k = Tensor();
+  auto grad_v = Tensor();
+
+  if (!grad_o.defined() && !grad_a.defined()) {
```

Comment:
pretty verbose `matmul` instead of `mm` but was trying to be as explicit as possible

Reply:
wdym `matmul` is less explicit than `mm` no?

https://pytorch.org/docs/stable/generated/torch.matmul.html
https://pytorch.org/docs/stable/generated/torch.mm.html

Reply:
Yeah I guess matmul doesn't conform to the explicit argument but using the full transpose(0,1) instead of t() and using mul() instead of * which I think is overloaded to element wise product